cmd/kubeadm/app/discovery/token/token.go

```diff
@@ -131,32 +104,11 @@ func RetrieveValidatedConfigInfo(cfg *kubeadmapi.JoinConfiguration) (*clientcmda
 			return nil, errors.Wrapf(err, "cluster CA found in %s configmap is invalid", bootstrapapi.ConfigMapClusterInfo)
 		}
 
-		// Now that we know the proported cluster CA, connect back a second time validating with that CA
-		secureBootstrapConfig := buildSecureBootstrapKubeConfig(endpoint, clusterCABytes, clusterName)
-		secureClient, err := kubeconfigutil.ToClientSet(secureBootstrapConfig)
-		if err != nil {
-			return nil, err
-		}
-
 		klog.V(1).Infof("[discovery] Requesting info from %q again to validate TLS against the pinned public key\n", insecureBootstrapConfig.Clusters[clusterName].Server)
-		var secureClusterInfo *v1.ConfigMap
-		wait.PollImmediateInfinite(constants.DiscoveryRetryInterval, func() (bool, error) {
-			var err error
-			secureClusterInfo, err = secureClient.CoreV1().ConfigMaps(metav1.NamespacePublic).Get(bootstrapapi.ConfigMapClusterInfo, metav1.GetOptions{})
-			if err != nil {
-				klog.V(1).Infof("[discovery] Failed to request cluster info, will try again: [%s]\n", err)
-				return false, nil
-			}
-			return true, nil
-		})
 
-		// Pull the kubeconfig from the securely-obtained ConfigMap and validate that it's the same as what we found the first time
-		secureKubeconfigBytes := []byte(secureClusterInfo.Data[bootstrapapi.KubeConfigKey])
-		if !bytes.Equal(secureKubeconfigBytes, insecureKubeconfigBytes) {
-			return nil, errors.Errorf("the second kubeconfig from the %s configmap (using validated TLS) was different from the first", bootstrapapi.ConfigMapClusterInfo)
-		}
-
-		secureKubeconfig, err := clientcmd.Load(secureKubeconfigBytes)
+		// Now that we know the proported cluster CA, connect back a second time validating with that CA
+		secureBootstrapConfig := buildSecureBootstrapKubeConfig(endpoint, clusterCABytes, clusterName)
```

Comment:
secureBootstrapConfig seems used only inside getSecureKubeconfig....

Reply:
secureBootstrapConfig used only inside getSecureKubeconfig, but if we put it in getSecureKubeconfig, we need to pass more parameters, so I put it out of getSecureKubeconfig.